docs/source/json-api/production-setup.rst

```diff
@@ -0,0 +1,300 @@
+.. Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+Production Setup
+################
+
+The vast majority of the prior documentation focuses on ease of testing and running
+the service in a dev environment. From a production perspective given the wide
+variety of use-cases there is far less of an established framework for deploying
+the *HTTP-JSON API* server. In this document we would try to list some recommendations for
+production deployments.
+
+The *HTTP-JSON API* server is a JVM application that by default uses an in-memory backend.
+This in-memory backend setup is inefficient for larger datasets as for every query it
+ends up fetching the entire active contract set for the templates referenced in that query.
+For this reason for production setups at a minimum we recommend to use a database
+as a query store, this will allow for more efficient caching of the data to improve
+query performance. Details for enabling a query store are highlighted below.
+
+Query Store
+***********
+
+Query store can be described as a cached search index and is useful for use-cases
+where we need to query large active contract sets. The *HTTP-JSON API* server can be configured
+with PostgreSQL/Oracle(Enterprise Edition only) as the query-store backend.
+
+For example to enable the PostgreSQL backend you can use the ``--query-store-jdbc-config`` flag, as shown below.
+
+.. code-block:: shell
+
+    daml json-api --ledger-host localhost --ledger-port 6865 --http-port 7575 \
+    --query-store-jdbc-config "driver=org.postgresql.Driver,url=jdbc:postgresql://localhost:5432/test?&ssl=true,user=postgres,password=password,start-mode=start-only"
+
+For using the Query Store you'll want your first run to specify ``start-mode=create-only``
+so that all the necessary tables are created. After the first run make
+sure its set to ``start-mode=start-only`` so that it doesn't attempt to create
```

Comment:
```suggestion
sure it is set to ``start-mode=start-only`` so that it doesn't attempt to create
```

Would it be worth having a small discussion of the pros and cons of various `start-mode` options? If this is already documented somewhere else, at least a link would be good. I'm personally not a fan of `create-only`, and would be a bit sad if that were our only recommended production setup.

Reply:
corrected this , based on the discussion I have added a couple of lines about how the `start-mode` can be used , please have a look to check if that makes sense @garyverhaegen-da :)

Reply:
👍